src/__tests__/project_workspace.test.js

```diff
@@ -11,9 +11,9 @@ import ProjectWorkspace from '../project_workspace';
 
 describe('setup', () => {
```

Comment:
That sounds like it might be fragile if we end up testing implementation details. Can we do that without knowing the internals?

Given a new ProjectWorkspace created with a jestCommandLine:
  - it should return the jestCommandLine
    ```js
    const commandLine = {} /* some arg */
    const instance = new ProjectWorkspace(..., commandLine, ...);
    const expected = commandLine;
    const actual = instance.jestCommandLine;

    expect(actual).toBe(expected);
    ```
  - it should continue to support the deprecated pathToJest
    ```js
    // The same setup with
    const actual = instance.jestCommandLine;

    expect(actual).toBe(expected);
    ```

Given a new ProjectWorkspace created with a pathToJest:
  - it should return the jestCommandLine
  - it should continue to support the deprecated pathToJest

Reply:
I had a test which exercised this functionality (at least IMO).  I moved it from the settings tests to this file.  Here is the definition:

```ts
it('ensure that pathToJest is a shadow accessor for jestCommandLine', () => {
    const localJestMajorVersion = 1000;
    const pathToConfig = 'test';
    const jestCommandLine = 'path_to_jest';
    const rootPath = 'root_path';
    const workspace = new ProjectWorkspace(rootPath, jestCommandLine, pathToConfig, localJestMajorVersion);

    // first check that both pathToConfig and jestCommandLine are the same value.
    expect(workspace.jestCommandLine).toBe(jestCommandLine);
    expect(workspace.pathToJest).toBe(jestCommandLine);

    // then update the pathToJest property.
    const updatedPathToJest = 'updated pathToJest';
    workspace.pathToJest = updatedPathToJest;

    // check that both pathToConfig and jestCommandLine yield the same value.
    expect(workspace.jestCommandLine).toBe(updatedPathToJest);
    expect(workspace.pathToJest).toBe(updatedPathToJest);
  });
```
Note that it is not possible to 'create' a config object with pathToJest rather than jestCommandLine because the constructor does not allow it (we should test this if/when we have a config _object_).  To test the shadow access, I have used property assignment instead.  Its unlikely anyone would create the object and then reassign the properties but it does prove the desired implementation.